dataclients/kubernetes/endpointslices.go

```diff
@@ -16,8 +16,10 @@ type skipperEndpointSlice struct {
 
 // Conditions have to be evaluated before creation
 type skipperEndpoint struct {
-	Address string
-	Zone    string
+	Address   string
+	Zone      string
+	NodeName  string
+	TargetRef *objectReference
```

Comment:
here we can and should only pass PodName and not ugly K8s references.

Reply:
Its package-internal struct and only used for unmarshaling k8s API data